pkg/groups/validation.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package groups
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/validation"
+)
+
+type PayloadValidator struct {
+	apiKey    string
+	validator *validator.Validate
+}
+
+func (_ *PayloadValidator) NeedsValidation(r *http.Request) bool {
```

Comment:
```suggestion
func (p *PayloadValidator) NeedsValidation(r *http.Request) bool {
```
unless linter complains I d leave the p, easier to understand it belongs (mainly a thing for grepping)